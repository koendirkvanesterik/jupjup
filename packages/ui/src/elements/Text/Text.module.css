.text {
  &--small {
    @apply text-sm;
  }
  &--medium {
  }
  &--large {
  }
}
