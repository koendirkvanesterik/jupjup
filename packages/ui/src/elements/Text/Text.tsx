import { createElement, PropsWithChildren } from 'react'
import styles from './Text.module.css'

type Props = PropsWithChildren<{
  component?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'span'
  size?: 'small' | 'medium' | 'large'
}>

export const Text = ({
  children,
  component = 'span',
  size = 'medium',
}: Props) =>
  createElement(component, { className: styles[`text--${size}`] }, children)
