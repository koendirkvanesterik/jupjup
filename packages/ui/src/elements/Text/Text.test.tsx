import { render } from '@testing-library/react'
import { Text, TextComponent, TextSize } from './Text'

it('should render as expected', () => {
  const { getByText } = render(<Text>Text</Text>)
  const text = getByText('Text')
  expect(text).toMatchSnapshot()
  expect(text).toHaveClass('text--medium')
})

it.each([
  ['h1' as TextComponent, 'heading'],
  ['h2' as TextComponent, 'heading'],
  ['h3' as TextComponent, 'heading'],
  ['h4' as TextComponent, 'heading'],
  ['h5' as TextComponent, 'heading'],
])('should render as `%s` element', (input, expected) => {
  const { getByRole } = render(<Text component={input} />)
  expect(getByRole(expected)).toBeInTheDocument()
})

it('should render as expected', () => {
  const { getByText } = render(<Text component="p">Text</Text>)
  const text = getByText('Text')
  expect(text).toMatchSnapshot()
})

it.each([
  ['small' as TextSize, 'text--small'],
  ['medium' as TextSize, 'text--medium'],
  ['large' as TextSize, 'text--large'],
])('should render with class name `text--%s`', (input, expected) => {
  const { getByRole } = render(<Text component="h1" size={input} />)
  expect(getByRole('heading')).toHaveClass(expected)
})
