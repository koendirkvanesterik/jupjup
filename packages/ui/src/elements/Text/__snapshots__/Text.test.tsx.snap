// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<span
  class="text--medium"
>
  Text
</span>
`;

exports[`should render as expected 2`] = `
<p
  class="text--medium"
>
  Text
</p>
`;
